play-framework/async-http/test/controllers/HomeControllerTest.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Test
+    public void givenASingleGetRequestWhenResponseThenBlockWithCompletableAndLog()
+      throws InterruptedException, ExecutionException {
+        AtomicInteger completedReqs = new AtomicInteger(0);
+        WSClient ws = play.test.WSTestClient.newClient(port);
+        WSResponse wsResponse = ws.url(url)
+                                  .setRequestFilter(new AhcCurlRequestLogger())
+                                  .addHeader("key", "value")
+                                  .addQueryParameter("num", "" + 1)
+                                  .get()
+                                  .toCompletableFuture()
+                                  .get();
+
+        log.debug("Thread#" + Thread.currentThread()
+                                    .getId() + " Request complete: Response code = "
+          + wsResponse.getStatus()
+          + " | Response: " + wsResponse.getBody() + " | Current Time:"
+          + System.currentTimeMillis());
+    }
+
+    @Test
+    public void givenASingleGetRequestWhenResponseThenLog() throws InterruptedException {
+        AtomicInteger completedReqs = new AtomicInteger(0);
+        WSClient ws = play.test.WSTestClient.newClient(port);
+        ws.url(url)
+          .setRequestFilter(new AhcCurlRequestLogger())
+          .addHeader("key", "value")
+          .addQueryParameter("num", "" + 1)
+          .get()
+          .thenAccept(r -> {
+              log.debug("Thread#" + Thread.currentThread()
+                                          .getId() + " Request complete: Response code = "
+                + r.getStatus()
+                + " | Response: " + r.getBody() + " | Current Time:" + System.currentTimeMillis());
+              completedReqs.incrementAndGet();
+          });
+
+        log.debug(
+          "Waiting for requests to be completed. Current Time: " + System.currentTimeMillis());
+        while (completedReqs.get() != 1) {
+            Thread.sleep(100);
+        }
+        log.debug("All requests have been completed. Exiting test.");
+    }
+
+    @Test
+    public void givenASinglePostRequestWhenResponseThenLog() throws InterruptedException {
+        AtomicInteger completedReqs = new AtomicInteger(0);
+        WSClient ws = play.test.WSTestClient.newClient(port);
+        ws.url(url)
+          .setContentType("application/x-www-form-urlencoded")
+          .post("key1=value1&key2=value2")
+          .thenAccept(r -> {
+              log.debug("Thread#" + Thread.currentThread()
+                                          .getId() + " Request complete: Response code = "
+                + r.getStatus()
+                + " | Response: " + r.getBody() + " | Current Time:" + System.currentTimeMillis());
+              completedReqs.incrementAndGet();
+          });
+
+        log.debug(
+          "Waiting for requests to be completed. Current Time: " + System.currentTimeMillis());
+        while (completedReqs.get() != 1) {
+            Thread.sleep(100);
+        }
```

Comment:
Another good use case for `CountDownLatch`

Reply:
done